rllib/connectors/util.py

```diff
@@ -78,8 +78,10 @@ def create_connectors_for_policy(policy: "Policy", config: TrainerConfigDict):
     """
     ctx: ConnectorContext = ConnectorContext.from_policy(policy)
 
-    policy.agent_connectors = get_agent_connectors_from_config(ctx, config)
-    policy.action_connectors = get_action_connectors_from_config(ctx, config)
+    if policy.agent_connectors is None:
```

Comment:
hmm, I wonder why we need to check this.
are we calling this function for an existing policy that already has connectors restored??

Reply:
What if you have recovered this policy in here from a checkpoint? Then you would also already have the connectors inside this policy, correct?
In this case, you wouldn't want to re-create the connectors. Let me know if this chain of thought is wrong.

Reply:
we shouldn't be calling this util if we are recovering a policy. this is only used when a policy is constructed from scratch.
do you mind removing these checks? if things fail somehow, I'd rather get an explicit signal than having actual problem concealed by these.

Reply:
Got it, so you are saying we can assume 100% that every time we are recovering a policy say from a checkpoint, the connectors should already be in there? In this case, I added an assert to the utility and fixed the `add_policy` method to NOT call this utility iff `policy` was provided as an already instantiated one.